Sources/protoc-gen-swift/MessageFieldGenerator.swift

```diff
@@ -194,31 +198,59 @@ class MessageFieldGenerator: FieldGeneratorBase, FieldGenerator {
             traitsArg = ""
         }
 
-        let varName = hasFieldPresence ? "v" : storedProperty
-
         var usesLocals = false
-        let conditional: String
+        let conditionals: [(condition: String, varName: String)]
         if isRepeated {  // Also covers maps
-            conditional = "!\(varName).isEmpty"
+            conditionals = [
+                (condition: "visitDefaultValues || !\(storedProperty).isEmpty", varName: storedProperty)
+            ]
         } else if hasFieldPresence {
-            conditional = "let v = \(storedProperty)"
             usesLocals = true
+            if !isGroupOrMessage, fieldDescriptor.file.syntax == .proto2 {
+                conditionals = [
+                    (condition: "let v = \(storedProperty)", varName: "v"),
+                    (condition: "visitDefaultValues", varName: swiftName),
```

Comment:
Does this need to be split to allow us the get the correct default value passed through since the stored value is nil at this point?  I wonder if we could avoid the two visit calls by crafting better generation here?  like can we make the `let v` somehow also get the default value (since the generator has it?  use `??` with a ternary?

Reply:
it would be done like this instead I think:
```swift
if visitDefaultValues || _someProperty != nil {
    try visitor.visitFloatSingularInt32(value: _someProperty ?? 0, fieldNumber: ...) 
}
```
This requires accessing `_someProperty` twice though so IDK if thats better or worse 🤷‍♂️

Reply:
I guess I was thinking something like:

```swift
if let v = _someProperty ?? ( visitDefaultValues ? 0 : nil ) {
    try visitor.visitFloatSingularInt32(value: v, fieldNumber: ...) 
}
```

(typed in github comment field, no claims it is remotely close to valid code)

Reply:
Oh interesting! I'll try this